### PR TITLE
Upgrade StyleCop

### PR DESCRIPTION
**Pull request by blairconrad**
Updated rules will work better with collection expressions when we start using them.

### CODE REVIEW

File: src/FakeItEasy/Core/CompletedFakeObjectCall.cs
Comment (blairconrad) [inline src/FakeItEasy/Core/CompletedFakeObjectCall.cs:29]: New StyleCop was snippy about the lack of newline.

**Comment by blairconrad** [comment]
Actually I think I see a [new warning from the build project](https://github.com/FakeItEasy/FakeItEasy/actions/runs/7365846031/job/20047530566?pr=1979#step:4:11). Which we probably can't address in here. I can try to address before we merge this if you like.

File: tools/FakeItEasy.Build/Program.cs
Comment (thomaslevesque) [inline tools/FakeItEasy.Build/Program.cs:122]: What do you mean? We can absolutely specify an access modifier.
Declaring it `public` causes `CA1050`, but `internal` or `file` works fine.
I think it would be better than suppressing the warning.

**Review by thomaslevesque — CHANGES_REQUESTED** [review]
Thanks @blairconrad. Just a small request!

File: tools/FakeItEasy.Build/Program.cs
Comment (blairconrad) [reply tools/FakeItEasy.Build/Program.cs:122]: I'm a moron. I tried `private`. Will amend.

File: Directory.Build.props
Comment (blairconrad) [reply Directory.Build.props:15]: If only they'd stop jiggling things.

**Review by thomaslevesque — APPROVED** [review]
LGTM, thanks!

**Comment by blairconrad** [comment]
Ugh. I left the `using`.

**Review by thomaslevesque — APPROVED** [review]
I think it's good this time ^^

**Comment by blairconrad** [comment]
Thanks for the pushback!

**Pull request merged**
